src/calibrationlib/calibration_benchmark_runner.cpp

```diff
@@ -0,0 +1,104 @@
+  auto config = std::make_shared<BenchmarkConfig>(BenchmarkConfig::get_default_config());
+  config->max_runs = 1;
+  config->enable_visualization = false;
+  config->chunk_size = 100'000;
+  config->cache_binary_tables = true;
+  _config = config;
+}
+
+CalibrationBenchmarkRunner::CalibrationBenchmarkRunner(const std::string& path_to_dir,
+                                                       std::shared_ptr<BenchmarkConfig> config)
+    : _feature_export(OperatorFeatureExport(path_to_dir)), _table_export(TableFeatureExport(path_to_dir)) {
+  _config = config;
+}
+
+void CalibrationBenchmarkRunner::run_benchmark(const BenchmarkType type, const float SCALE_FACTOR,
+                                               const int number_of_executions) {
+  std::shared_ptr<BenchmarkRunner> benchmark_runner;
+  switch (type) {
+    case BenchmarkType::TCPH:
+      benchmark_runner = _build_tcph(SCALE_FACTOR);
+      break;
+    case BenchmarkType::TCPDS:
+      benchmark_runner = _build_tcpds(SCALE_FACTOR);
+      break;
+    case BenchmarkType::JOB:
+      benchmark_runner = _build_job();
+      break;
+    default:
+      throw std::runtime_error("Provided unknown BenchmarkType.");
+  }
+
+  for (int i = 0; i < number_of_executions; ++i) {
+    Hyrise::get().benchmark_runner = benchmark_runner;
+    benchmark_runner->run();
+
+    auto& pqp_cache = Hyrise::get().default_pqp_cache;
+
+    for (auto pqp_entry = pqp_cache->unsafe_begin(); pqp_entry != pqp_cache->unsafe_end(); ++pqp_entry) {
+      const auto& [query_string, physical_query_plan] = *pqp_entry;
+      _feature_export.export_to_csv(physical_query_plan);
+    }
+
+    // Clear pqp cache for next benchmark
+    pqp_cache->clear();
+  }
+
+  const std::vector<std::string> table_names = Hyrise::get().storage_manager.table_names();
+  for (const auto& table_name : table_names) {
+    auto table = Hyrise::get().storage_manager.get_table(table_name);
+    _table_export.export_table(std::make_shared<CalibrationTableWrapper>(CalibrationTableWrapper(table, table_name)));
+
+    // Drop table because we do not need it anymore
+    Hyrise::get().storage_manager.drop_table(table_name);
+  }
+}
+
+std::shared_ptr<BenchmarkRunner> CalibrationBenchmarkRunner::_build_tcph(const float SCALE_FACTOR) const {
+  auto item_runner = std::make_unique<TPCHBenchmarkItemRunner>(_config, false, SCALE_FACTOR);
+  auto benchmark_runner = std::make_shared<BenchmarkRunner>(*_config, std::move(item_runner),
+                                                            std::make_unique<TPCHTableGenerator>(SCALE_FACTOR, _config),
+                                                            BenchmarkRunner::create_context(*_config));
+
+  return benchmark_runner;
+}
+
+std::shared_ptr<BenchmarkRunner> CalibrationBenchmarkRunner::_build_tcpds(const float SCALE_FACTOR) const {
+  const std::string query_path = "hyrise/resources/benchmark/tpcds/tpcds-result-reproduction/query_qualification";
```

Comment:
That path does not seem right. We usually execute the binaries from the _hyrise_ folder.

Reply:
So, `resources/benchmark/tpcds/tpcds-result-reproduction/query_qualification` should be correct, right?

Reply:
The path has been adapted in the plugin, because the plugin includes Hyrise as a third-party module. Is it really necessary to pass the path here?

Reply:
Yes, it is, but it doesn't matter for this PR anymore because I removed TPC-DS.